src/envoy/http/service_control/client_cache_test.cc

```diff
@@ -148,6 +148,52 @@ TEST_F(ClientCacheCheckResponseNetworkFailClosedTest, Sc5xxBlocked) {
   checkAndReset(stats_base_.stats().filter_.denied_control_plane_fault_, 1);
 }
 
+class ClientCacheCheckResponseErrorTypeTest
+    : public ClientCacheCheckResponseTest {
+ protected:
+  void runTest(CheckResponse* got_response) {
+    CheckDoneFunc on_done = [&](const Status&, const CheckResponseInfo&) {};
+    const Status http_status(Code::OK, Envoy::EMPTY_STRING);
+    cache_->handleCheckResponse(http_status, got_response, on_done);
+  }
+};
+
+TEST_F(ClientCacheCheckResponseErrorTypeTest, ConsumerBlocked) {
+  CheckResponse* response = new CheckResponse();
+  CheckError* check_error = response->mutable_check_errors()->Add();
+  check_error->set_code(CheckError::CLIENT_APP_BLOCKED);
```

Comment:
it seems that you could move above 3 lines into runTest()

Reply:
Done